lib/events.js

```
@@ -0,0 +1,80 @@
+var amoeba = require('amoeba');
+var config = require('amoeba/lib/config')
+
+module.exports = async function (config, eventsLogger) {
+
+    function assertConfigValid(config) {
+        if (config == null) {
+          throw new Error('Config is not set');
+        }
+        if (!config.brokers || !config.brokers.length) {
+          throw new Error('Brokers is not set');
+        }
+        if (!config.cloudEventsSource) {
+          throw new Error('Cloud events source is not set');
+        }
+        if (!config.consumerGroup) {
+          throw new Error('Consumer group is not set');
+        }
+        if (!config.topic) {
+          throw new Error('Topic is not set');
+        }
+    }
+    
+    function getLogLevel(config) {
+        const level = logLevel[config.clientLogLevel];
+        return level || logLevel.ERROR;
+    }
+    
+    const opts = {
+        clientId: config.cloudEventsSource,
+        brokers: config.brokers,
+        logLevel: getLogLevel(config),
+        ssl: config.requireSSL,
+    };
+    if (config.username != null && config.password != null) {
+        opts.sasl = {
+            mechanism: 'scram-sha-512',
+            username: config.username,
+            password: config.password,
+        };
+    }
+
+    const kafka = new Kafka(opts);
+    const producer = kafka.producer()
+    await producer.connect().then(console.log("producer is connected")).catch(console.error)
+    return {
+        createEventsHandler: async function (event) {
+            const now = new Date()
+            const run = async () => {
+                assertConfigValid(config)
+                await producer.send({
+                    topic: config.topic,
+                    messages: [
+                        {value: JSON.stringify(event)},
+                    ],
+                }).then(console.log(now + " Kafka Message sent: " + JSON.stringify(event)))
+            }
+            run().catch(console.error)
```

Don't catch here, let it bubble up.